inference-engine/src/transformations/src/transformations/common_optimizations/nop_elimination.cpp

```diff
@@ -336,20 +358,168 @@ NAME() { \
+            std::any_of(pad_end_value.begin(), pad_end_value.end(), [](int64_t value) { return value != 0; })) {
+            return false;
+        }
+
+        return replace_output_update_name(pad->output(0), pad->input_value(0));
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(pad_node_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+NGRAPH_RTTI_DEFINITION(pass::EliminateConvert, "EliminateConvert", 0);
 
-ngraph::pass::NopElimination::NopElimination() {
+pass::EliminateConvert::EliminateConvert() {
+    MATCHER_SCOPE(EliminateConvert);
+    auto convert_pattern = pattern::wrap_type<opset8::Convert>();
+
+    matcher_pass_callback callback = [](pattern::Matcher& m) {
+        auto convert = std::dynamic_pointer_cast<opset8::Convert>(m.get_match_root());
+        if (!convert) {
+            return false;
+        }
+        if (convert->get_input_element_type(0) == convert->get_element_type()) {
+            return replace_output_update_name(convert->output(0), convert->input_value(0));
+        }
+        return false;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(convert_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+NGRAPH_RTTI_DEFINITION(pass::EliminateConvertNonZero, "EliminateConvertNonZero", 0);
+
+pass::EliminateConvertNonZero::EliminateConvertNonZero() {
+    MATCHER_SCOPE(EliminateConvertNonZero);
+    auto convert_pattern = pattern::wrap_type<opset8::Convert>(pattern::consumers_count(1));
+    auto non_zero = pattern::wrap_type<opset8::NonZero>({convert_pattern});
+
+    matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        const auto & pattern_map = m.get_pattern_map();
+        auto convert = pattern_map.at(convert_pattern);
+        // remove convert
+        convert->output(0).replace(convert->input_value(0));
+        // to make this elimination recursive we register NonZero as a node which will be used to repeat matching
+        register_new_node(m.get_match_root());
+        return true;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(non_zero, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+NGRAPH_RTTI_DEFINITION(pass::EliminateConcat, "EliminateConcat", 0);
+
+pass::EliminateConcat::EliminateConcat() {
+    MATCHER_SCOPE(EliminateConcat);
+    auto convert_pattern = pattern::wrap_type<opset8::Concat>();
+
+    matcher_pass_callback callback = [](pattern::Matcher& m) {
+        auto concat = m.get_match_root();
+        if (concat->inputs().size() == 1) {
+            return replace_output_update_name(concat->output(0), concat->input_value(0));
+        }
+        return false;
+    };
+
+    auto m = std::make_shared<pattern::Matcher>(convert_pattern, matcher_name);
+    this->register_matcher(m, callback);
+}
+
+NGRAPH_RTTI_DEFINITION(pass::EliminateSplit, "EliminateSplit", 0);
+
+pass::EliminateSplit::EliminateSplit() {
+    MATCHER_SCOPE(EliminateConcat);
```

Comment:
```suggestion
    MATCHER_SCOPE(EliminateSplit);
```

Reply:
If you don't mind I will fix it within https://github.com/openvinotoolkit/openvino/pull/6916 PR.